### PR TITLE
Hotfix network validation of CCE cluster

### LINKED ISSUE #803: Error: can't find VPC ``: Resource not found
# Terraform Version
`Terraform v0.14.5`
`opentelekomcloud/opentelekomcloud v1.22.4`

# Affected Resource(s)
Probably:
- `opentelekomcloud_cce_cluster_v3.cce`
- `opentelekomcloud_vpc_v1`

# Terraform Configuration Files
Very basic setup
```
resource "opentelekomcloud_vpc_v1" "anyname" {
  name = "anyname"
 ...
}

resource "opentelekomcloud_vpc_subnet_v1" "anyname" {
  vpc_id        = opentelekomcloud_vpc_v1.anyname.id
  name          = opentelekomcloud_vpc_v1.anyname.name
  ...
}

resource "opentelekomcloud_cce_cluster_v3" "anycce" {
  ...
  vpc_id      = opentelekomcloud_vpc_v1.anyname.id
  subnet_id   = opentelekomcloud_vpc_subnet_v1.anyname.id
  ...
}
```

# Debug Output/Panic Output

`Error: can't find VPC ``: Resource not found: [GET https://vpc.eu-de.otc.t-systems.com/v1//vpcs/], error message: {"message":"API not found","request_id":""}`

# Steps to Reproduce
1. `terraform init`
1. `terraform plan -target opentelekomcloud_cce_cluster_v3.anycce`

# Expected Behavior
Terraform should connect the VPC and the CEE properly as it did in `opentelekomcloud/opentelekomcloud v1.22.3`

# Actual Behavior
VPC name/identifier is not transmitted?

# Important Factoids

# References

This PR introduces the new error message: https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/pull/787

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request

Resolves #803 

## PR Checklist

* [x] Refers to: #803
* [x] Tests added/passed.

## Acceptance Steps Performed

```
=== RUN   TestAccCCEClusterV3_invalidNetwork
--- PASS: TestAccCCEClusterV3_invalidNetwork (19.33s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/customize_diffs.go [hunk 1]
```diff
@@ -38,3 +40,32 @@ func validateRDSv3Version(argumentName string) schema.CustomizeDiffFunc {
...
+	if !ok {
+		return fmt.Errorf("error retreiving configuration: can't convert %v to Config", meta)
+	}
+	vpcClient, err := config.networkingV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating opentelekomcloud CCE Client: %s", err)
+	}
+
+	vpcID := d.Get("vpc_id").(string)
+	if len(vpcID) != 0 {
```
Comment (outcatcher) [inline opentelekomcloud/customize_diffs.go]: `vpcID` can be empty string, you can make it just `if vpcID != ""`

File: opentelekomcloud/customize_diffs.go [hunk 2]
```diff
@@ -38,3 +40,32 @@ func validateRDSv3Version(argumentName string) schema.CustomizeDiffFunc {
...
+		err = vpcs.Get(vpcClient, vpcID).Err
+		if err != nil {
+			return fmt.Errorf("can't find VPC `%s`: %s", vpcID, err)
+		}
+	}
+
+	subnetID := d.Get("subnet_id").(string)
+	if len(subnetID) != 0 {
```
Comment (outcatcher) [inline opentelekomcloud/customize_diffs.go]: Same `!= ""` can be used

File: opentelekomcloud/customize_diffs.go [hunk 3]
```diff
@@ -38,3 +40,32 @@ func validateRDSv3Version(argumentName string) schema.CustomizeDiffFunc {
...
+	}
+	vpcClient, err := config.networkingV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating opentelekomcloud CCE Client: %s", err)
+	}
+
+	vpcID := d.Get("vpc_id").(string)
+	if len(vpcID) != 0 {
+		err = vpcs.Get(vpcClient, vpcID).Err
+		if err != nil {
```
Comment (outcatcher) [inline opentelekomcloud/customize_diffs.go]: You can use single-line expression:
`if err := vpcs.Get(vpcClient, vpcID).Err; if err != nil {`

File: opentelekomcloud/customize_diffs.go (hunk 1 above)
Comment (outcatcher) [inline opentelekomcloud/customize_diffs.go]: This can be one-line expression:
`if vpcID := d.Get("vpc_id").(string); vpcID != "" {...`

**Comment by outcatcher** [comment]
recheck

**Review by outcatcher — DISMISSED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**